assets/javascripts/kitten/components/typography/markdown/markdown-page.js

```diff
@@ -0,0 +1,197 @@
+import { Paragraph } from '../../../components/typography/paragraph'
+import { BulletList } from '../../../components/lists/bullet-list'
+import { LinkList } from '../../../components/links/link-list'
+import { HorizontalStroke } from '../../../components/layout/horizontal-stroke'
+import { scrollTo } from '../../../helpers/utils/scroll-to'
+import { pxToRem } from '../../../helpers/utils/typography'
+import { ScreenConfig } from '../../../constants/screen-config'
+
+export class MarkdownPage extends Component {
+  render() {
+    return (
+      <Markdown
+        source={this.props.content}
+        renderers={markdownRenderers}
+        escapeHtml={false}
+      />
+    )
+  }
+}
+
+MarkdownPage.propTypes = {
+  content: PropTypes.string.isRequired,
+}
+
+const MarkdownParagraph = props => {
+  return (
+    <Marger bottom="2.6">
+      <Paragraph modifier={props.modifierParagraph} margin={false}>
+        {props.children}
+      </Paragraph>
+    </Marger>
+  )
+}
+
+MarkdownParagraph.propTypes = {
+  children: PropTypes.node.isRequired,
+  modifierParagraph: PropTypes.string,
+}
+
+MarkdownParagraph.defaultProps = {
+  modifierParagraph: 'primary',
+}
+
+const MarkdownBlockquote = props => {
+  const items = props.children.map(item => ({
+    key: item.key,
+    item: item.props.children[0],
+  }))
+
+  return (
+    <Marger top="4" bottom="4">
+      <LinkListstyle color="dark" items={items} />
+    </Marger>
+  )
+}
+
+MarkdownBlockquote.propTypes = {
+  children: PropTypes.node.isRequired,
+}
+
+const MarkdownList = props => {
+  const items = props.children.map(item => ({
+    key: item.key,
+    item: item.props.children[0],
+  }))
+
+  return (
+    <Marger top="3" bottom="3">
+      <BulletListStyle big items={items} />
+    </Marger>
+  )
+}
+
+MarkdownList.propTypes = {
+  children: PropTypes.node.isRequired,
+}
+
+const handleLinkClick = url => {
+  if (url.match(/^#/)) {
+    scrollTo(document.getElementById(url.replace('#', '')), 500)
+  }
+}
+
+const MarkdownLink = props => {
+  return (
+    <a
+      href={props.href}
+      onClick={() => {
+        handleLinkClick(props.href)
+      }}
+    >
+      {props.children}
+    </a>
+  )
+}
+
+MarkdownLink.propTypes = {
+  href: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+}
+
+const MarkdownHeading = props => {
+  let modifier
+  let tag
+
+  switch (props.level) {
+    case 1:
+      modifier = 'secondary'
+      tag = 'h1'
+      break
+    case 2:
+      modifier = 'quaternary'
+      tag = 'h2'
+      break
+    case 3:
+      modifier = 'quinary'
+      tag = 'h3'
+      break
+    default:
+      modifier = 'senary'
+      tag = 'h4'
+      break
+  }
+
+  const renderHorizontalStroke = () => {
+    if (props.level > 1) return
+
+    return (
+      <Marger top="2" bottom="4">
+        <HorizontalStroke size="big" />
+      </Marger>
+    )
+  }
+
+  const id = slugify(props.children[0], {
+    lower: true,
+    remove: /[']/g,
```

Comment:
On peut simplifier cette expression régulière en :

```suggestion
    remove: /'/g,
```